src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+import ro.chirila.programarispital.repository.entity.User;
+import ro.chirila.programarispital.service.AppointmentService;
+import ro.chirila.programarispital.service.UserService;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+@Service
+public class AppointmentServiceImpl implements AppointmentService {
+
+    private final AppointmentRepository appointmentRepository;
+    private final ModelMapper modelMapper;
+    private final UserRepository userRepository;
+    private final TypeOfServiceRepository typeOfServiceRepository;
+
+    public AppointmentServiceImpl(AppointmentRepository appointmentRepository, ModelMapper modelMapper, UserRepository userRepository, TypeOfServiceRepository typeOfServiceRepository) {
+        this.appointmentRepository = appointmentRepository;
+        this.modelMapper = modelMapper;
+        this.userRepository = userRepository;
+        this.typeOfServiceRepository = typeOfServiceRepository;
+    }
+
+    @Override
+    public AppointmentResponseDTO addAppointment(AppointmentRequestDTO appointment, String username) {
+        validateFields(appointment);
+        Appointment savedAppointment = new Appointment();
+
+        User user = userRepository.findByUsername(username).orElseThrow(() -> new UserNotFoundException("No user found by this username"));
+        savedAppointment.setScheduledPerson(user);
+        savedAppointment.setEmail(appointment.getEmail());
+        savedAppointment.setFirstName(appointment.getFirstName());
+        savedAppointment.setLastName(appointment.getLastName());
+        savedAppointment.setPhoneNumber(appointment.getPhoneNumber());
+        savedAppointment.setDateOfBirth(new Date());
+        savedAppointment.setChooseDate(new Date());
+        savedAppointment.setAppointmentHour(appointment.getAppointmentHour());
+        savedAppointment.setPeriodOfAppointment(appointment.getPeriodOfAppointment());
+        savedAppointment.setTypeOfServices(new ArrayList<>());
+
+        for(TypeOfServiceDTO typeOfServiceDTO : appointment.getTypeOfServices()){
+            TypeOfService typeOfService = typeOfServiceRepository.findByService(modelMapper.map(typeOfServiceDTO, TypeOfService.class).getService());
+
+            if(typeOfService == null){
+                savedAppointment.getTypeOfServices().add(modelMapper.map(typeOfServiceDTO, TypeOfService.class));
+            }else{
+                savedAppointment.getTypeOfServices().add(typeOfService);
+            }
+        }
+
+        if(user.getAppointments() == null){
+            user.setAppointments(new ArrayList<>());
+        }
+        user.getAppointments().add(savedAppointment);
+
+        AppointmentResponseDTO responseDTO = modelMapper.map(appointmentRepository.save(savedAppointment), AppointmentResponseDTO.class);
+        responseDTO.setScheduledPerson(user);
+        return responseDTO;
+
+    }
+
+    public void validateFields(AppointmentRequestDTO appointment){
```

Comment:
I recommend doing validation using bean validation like this: https://www.baeldung.com/spring-boot-bean-validation

Reply:
Fixed